packages/pds/src/services/repo/index.ts

```diff
@@ -229,43 +234,76 @@ export class RepoService {
     await sequencer.sequenceEvt(this.db, seqEvt)
   }
 
+  // rebases are expensive & should be done rarely, we don't try to re-process on concurrent writes
   async rebaseRepo(did: string, swapCommit?: CID) {
-    this.db.assertNotTransaction()
-    const storage = new SqlRepoStorage(this.db, did)
-    const currRoot = await storage.getHead()
-    if (!currRoot) {
-      throw new InvalidRequestError(
-        `${did} is not a registered repo on this server`,
-      )
-    }
-    const repo = await Repo.load(storage, currRoot)
-    const rebaseData = await repo.formatRebase(this.repoSigningKey)
-
-    // rebases are expensive & should be done rarely, we don't try to re-process on concurrent writes
-    await this.serviceTx(async (srvcTx) =>
-      srvcTx.processRebase(did, rebaseData, swapCommit),
-    )
```

Comment:
Is it easy to explain why we switched to do the formatting during the transaction?

Reply:
basically the lock was my intention. since we don't have any retry logic & want to lock the repo for the duration of the rebase.

that being said, we could still format the rebase ahead of time 🤔

Reply:
actually yknow probably better to do this outside of a txn after all